### DISCUSSION TITLE
relationship select column from another table

### DISCUSSION THREAD

**Discussion by riwu** [sqlalchemy/sqlalchemy#10715] [Usage Questions]
```python
class Country(Base):
    __tablename__ = "country"
    id = Column(Integer, primary_key=True)


class City(Base):
    __tablename__ = "city"

    id = Column(Integer, primary_key=True)
    country_id = Column(Integer, ForeignKey(Country.id), nullable=False)


class TaxRate(Base):
    __tablename__ = "tax_rate"

    country_id = Column(Integer, ForeignKey(Country.id), primary_key=True)
    income_bracket = Column(Text, primary_key=True)
    rate = Column(Integer)


class Person(Base):
    __tablename__ = "person"

    id = Column(Integer, primary_key=True)
    city_id = Column(Integer, ForeignKey(City.id), nullable=False)
    income_bracket = Column(Text, nullable=False)

    tax_rate: TaxRate = relationship(
        TaxRate,
        secondary="city",
        secondaryjoin=City.id == city_id,
        primaryjoin=and_(
            TaxRate.country_id == City.country_id,
            TaxRate.income_bracket == income_bracket,
        ),
    )
```

In the above schema, I want to define a `tax_rate` relationship in `Person`.
  
`TaxRate` has a composite PK (`country_id`, `income_bracket`).  
However, `Person` has only `city_id` so it needs to `JOIN` with `City` to get the `country_id`).

What is the correct syntax to define this relationship?  
I get this error with the above code:

    sqlalchemy.exc.ProgrammingError: (psycopg2.errors.UndefinedTable) missing FROM-clause entry for table "person"

**Comment by CaselIT** [comment]
Hi,

This seems similar to this one https://github.com/sqlalchemy/sqlalchemy/discussions/10208 and the other linked ones. There is a suggestion that you may try there

**Reply by zzzeek** [reply]
specifically https://github.com/sqlalchemy/sqlalchemy/discussions/10130#discussioncomment-6509006 which will point you to the patterns for this, see https://docs.sqlalchemy.org/en/20/orm/join_conditions.html#composite-secondary-joins and https://docs.sqlalchemy.org/en/20/orm/join_conditions.html#relationship-to-aliased-class
